src/ctia/lib/kafka.clj

```diff
@@ -0,0 +1,123 @@
+(ns ctia.lib.kafka
+  (:require
+   [clojure.tools.logging :as log]
+   [ctia.properties :refer [properties]]
+   [onyx.kafka.helpers :as okh]
+   [onyx.plugin.kafka :as opk])
+  (:import kafka.admin.AdminUtils))
+
+(defn decompress
+  [v]
+  (when v (String. v "UTF-8")))
+
+(defn ssl-enabled? []
+  (boolean (get-in @properties
+                   [:ctia :hook :kafka :ssl :enabled])))
+
+(defn make-ssl-opts []
+  (let [ssl-props (get-in @properties [:ctia :hook :kafka :ssl])]
+    {"security.protocol" "ssl"
+     "ssl.truststore.location"
+     (get-in ssl-props [:truststore :location])
+     "ssl.truststore.password"
+     (get-in ssl-props [:truststore :password])
+     "ssl.keystore.location"
+     (get-in ssl-props [:keystore :location])
+     "ssl.keystore.password"
+     (get-in ssl-props [:keystore :password])
+     "ssl.key.password"
+     (get-in ssl-props [:key :password])}))
+
+(defn build-producer []
+  (let [producer-opts {}
+        {:keys [request-size compression]}
+        (get-in @properties [:ctia :hook :kafka])
+        compression-type (:type compression)
+        {:keys [session-timeout
+                connection-timeout
+                operation-retry-timeout
+                address] :as zk-config}
+        (get-in @properties [:ctia :hook :kafka :zk])
+        _ (clojure.pprint/pprint compression-type)
+        brokers (opk/find-brokers {:kafka/zookeeper address})
+        kafka-config (cond-> {"bootstrap.servers" brokers
+                              "max.request.size" request-size}
+                       (ssl-enabled?) (merge (make-ssl-opts))
```

Comment:
```suggestion
                       (ssl-enabled?) (into (make-ssl-opts))
```

When we only have two maps, we usually use `into` instead of `merge`